### DISCUSSION TITLE
Welcome to life_code Discussions!

### DISCUSSION THREAD

**Discussion by hunar4321** [hunar4321/particle-life#16] [Announcements]
## 👋 Welcome!

  * Ask questions you’re wondering about.
  * Share ideas.
  * Share interesting findings.

**Comment by M16A2SA** [comment]
Is it possible to add a feature so that we can simply edit the numbers rather than dragging the sliders, this way it's easier to get control of everything in my opinion. :) (It's not very important though.)

**Reply by hunar4321** [reply]
> Is it possible to add a feature so that we can simply edit the numbers rather than dragging the sliders, this way it's easier to get control of everything in my opinion. :) (It's not very important though.)

Actually, at the beginning I was using number inputs instead of sliders, however, openFramework's library doesn't update edited numbers in real-time i.e you have to push reset to see the changes. For sliders the changes happen in real-time and in my opinion this real-time visual feedback is essential for fine-tuning & finding interesting patterns.  We'll revisit this option in later releases.

**Reply by M16A2SA** [reply]
I just had a glance on the openFramework's docs and I have an idea: You can check whether the user has completed the input through `ofKeyEventArgs()`, then update the numbers. (It's just an immature idea)
I am unfamiliar to both C++ and openFramework, so please forgive all misunderstandings!

**Comment by hunar4321** [comment]
I have added a simple Python port to this repo.

**Reply by LuizMosciaro** [reply]
Youre the man!
Congrats on the YouTube video, perfect on every point.
I'll study more this project, hope i might be usefull someway for you guys.

**Comment by hunar4321** [comment]
Here is a live online demo for the JavaScript file: https://hunar4321.github.io/particle-life/particle_life.html#91651088029

**Comment by hunar4321** [comment]
I have added the ability to save and load parameters for the C++ version and made the screen size 1600 by 900 to be a bit more friendly for those who don't have Full HD monitor.  I also added a folder where I'm planning to add some interesting models showcased in the video so that users can easily load these models. In future, we could add many more interesting models.

**Comment by Himbeereis** [comment]
Hi, I'm just a beginner who is trying to compile your cpp source code. When I try to compile I get "fatal error: 'ofMain.h' file not found". Could you tell me why I get this error? Thank you

**Reply by ker2x** [reply]
you need to install openframeworks then launch the project generator and import the cpp code

**Comment by GabeFirst** [comment]
I was wondering if the gravity in Life_code works linear. Because it should be an accelerated movment :). I looks to me, as if the game does not use acceleration, could be wrong tho

**Reply by ker2x** [reply]
```cpp
const float force = distance_squared < radius* radius ? 1.0F / std::sqrtf(distance_squared) : 0.0F;
fx += ((Group1.pos[i].x - Group2.pos[j].x) * force);
fy += ((Group1.pos[i].y - Group2.pos[j].y) * force);
```

If I remember the cpp code part correctly, It "use acceleration".
However, again if I remember correctly (it's been a long time), the particles are massless, therefore there is no inertia.
